src/cloud-api-adaptor/pkg/securecomms/sshproxy/sshproxy.go

```diff
@@ -0,0 +1,544 @@
+package sshproxy
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/securecomms/sshutil"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	PP_SID         = "pp-sid/"
+	PP_PRIVATE_KEY = PP_SID + "privateKey"
+	ATTESTATION    = "Attestation"
+	KUBERNETES     = "Kubernetes"
+)
+
+var logger = sshutil.Logger
+
+type SshPeer struct {
+	sid            string
+	phase          string
+	terminated     string
+	sshConn        ssh.Conn
+	ctx            context.Context
+	done           chan bool
+	outbounds      map[string]*Outbound
+	inbounds       map[string]*Inbound
+	wg             sync.WaitGroup
+	upgrade        bool
+	outboundsReady chan bool
+}
+
+// Inbound side of the Tunnel - incoming tcp connections from local clients
+type Inbound struct {
+	// tcp peers
+	Name        string
+	TcpListener *net.TCPListener
+	Connections chan *net.Conn
```

Comment:
Similar comment as - https://github.com/confidential-containers/cloud-api-adaptor/pull/1776/files#r1580524336
Defining string constant for the phases will be helpful imho.